storey/aggregations.py

```diff
@@ -189,14 +190,16 @@ def __init__(self, cache):
         super().__init__()
         self._cache = cache
 
-    async def _do(self, event):
-        if event is _termination_obj:
-            await self._cache.close()
-            return await self._do_downstream(_termination_obj)
-        else:
-            # todo: persist keys in parallel
-            await self._cache._persist_key(event.key)
-            await self._do_downstream(event)
+    async def _process_event(self, event):
+        return await self._cache.persist_key(event.key)
+
+    async def _handle_completed(self, event, response):
+        await self._do_downstream(event)
+
+    def run(self):
+        closeables = super().run()
+        closeables.append(self._cache)
+        return closeables
```

Comment:
Probably easier to have each step declare its closables and have run do the append.